src/Security/SecurityConfigUpdater.php

```diff
@@ -55,24 +75,47 @@ public function updateForAuthenticator(string $yamlSource, string $firewallName,
             $newData['security']['firewalls'] = [];
         }
 
-        if (!isset($newData['security']['firewalls'][$firewallName])) {
+        if (!isset($newData['security']['firewalls'][$firewallName]) && !$useSecurity51) {
             $newData['security']['firewalls'][$firewallName] = ['anonymous' => true];
+        } elseif (!isset($newData['security']['firewalls'][$firewallName]) && $useSecurity51) {
+            $newData['security']['firewalls'][$firewallName] = ['lazy' => true];
         }
```

Comment:
Logically it might read better as:

```php
if (!isset($newData['security']['firewalls'][$firewallName])) {
    if ($useSecurity51) {

    } else {

    }
}
```

Also, it's minor, but I think we were already missing that `lazy` should be included in ALL firewalls, at least back to Symfony 4.4. And... I guess we should check the version of Symfony to see if `lazy` exists? Or, we (in a separate PR) could drop support for 3.4 - bump it to 4.4, as 3.4 is no longer supported.

Reply:
`lazy: true` is introduced in 5.1, it used to be `anonymous: lazy`.

Reply:
done